src/createAnimatedComponent.tsx

```diff
@@ -409,11 +493,66 @@ export default function createAnimatedComponent(
       }
     }
 
+    _attachInlineStyles() {
+      const newInlineStyles: StyleProps = getInlineStylesFromProps(this.props);
+      const hasChanged = inlineStylesHasChanged(
+        newInlineStyles,
+        this._inlineStyles
+      );
+
+      if (hasChanged) {
+        if (!this._inlineStylesViewDescriptors) {
+          this._inlineStylesViewDescriptors = makeViewDescriptorsSet();
+
+          const { viewTag, viewName, shadowNodeWrapper } = this._getViewInfo();
+
+          this._inlineStylesViewDescriptors.add({
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            tag: viewTag!,
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            name: viewName!,
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            shadowNodeWrapper: shadowNodeWrapper!,
+          });
+        }
+        const sharableViewDescriptors =
+          this._inlineStylesViewDescriptors.sharableViewDescriptors;
+
+        const maybeViewRef = NativeReanimatedModule.native
+          ? undefined
+          : ({ items: new Set([this]) } as ViewRefSet<any>); // see makeViewsRefSet
+
+        const updaterFunction = () => {
+          'worklet';
+          const update = getInlineStyleUpdate(newInlineStyles);
+          updateProps(sharableViewDescriptors, update, maybeViewRef);
+        };
+        this._inlineStyles = newInlineStyles;
+        if (this._inlineStylesMapperId) {
+          stopMapper(this._inlineStylesMapperId);
+        }
+        this._inlineStylesMapperId = null;
+        if (Object.keys(newInlineStyles).length) {
+          this._inlineStylesMapperId = startMapper(
+            updaterFunction,
+            Object.values(newInlineStyles)
```

Comment:
it seem like if there is a transform object with only one of transform elements being a shared value, this won't actually extract that shared value. Can you verify if this works ok?